vividus-to-zephyr-exporter/src/main/java/org/vividus/zephyr/facade/ZephyrFacade.java

```diff
@@ -0,0 +1,148 @@
+
+        String cycleId = findCycleId(projectAndVersionUrlQuery);
+        zephyrConfiguration.setCycleId(cycleId);
+
+        String folderId = findFolderId(cycleId, projectAndVersionUrlQuery);
+        zephyrConfiguration.setFolderId(folderId);
+
+        Map<TestCaseStatus, Integer> statusIdMap = getExecutionStatuses();
+        zephyrConfiguration.setStatusIdMap(statusIdMap);
+
+        return zephyrConfiguration;
+    }
+
+    private String findVersionId(Project project)
+    {
+        Optional<Version> version = project.getVersions().stream().filter(
+            v -> zephyrExporterConfiguration.getVersionName().equals(v.getName())).findFirst();
+        isTrue(version.isPresent(), "Version with name '%s' does not exist",
+                zephyrExporterConfiguration.getVersionName());
+        return version.get().getId();
+    }
+
+    private String findCycleId(String projectAndVersionUrlQuery) throws IOException
+    {
+        String json = client.executeGet(ZAPI_ENDPOINT + "cycle?" + projectAndVersionUrlQuery);
+        Map<String, Map<String, String>> cycles = JsonPathUtils.getData(json, "$");
+        cycles.remove("recordsCount");
+        Iterator<Map.Entry<String, Map<String, String>>> itr = cycles.entrySet().iterator();
+        String cycleId = "";
+        while (itr.hasNext())
+        {
+            Map.Entry<String, Map<String, String>> map = itr.next();
+            if (map.getValue().get("name").equals(zephyrExporterConfiguration.getCycleName()))
+            {
+                cycleId = map.getKey();
+                break;
+            }
+        }
+        notBlank(cycleId, "Cycle with name '%s' does not exist", zephyrExporterConfiguration.getCycleName());
+        return cycleId;
+    }
+
+    private String findFolderId(String cycleId, String projectAndVersionUrlQuery) throws IOException
+    {
+        String json = client.executeGet(ZAPI_ENDPOINT + "cycle/" + cycleId + "/folders?" + projectAndVersionUrlQuery);
+        List<Integer> folderId = JsonPathUtils.getData(json, String.format("$.[?(@.folderName=='%s')].folderId",
+                zephyrExporterConfiguration.getFolderName()));
+        notEmpty(folderId, "Folder with name '%s' does not exist",
+                zephyrExporterConfiguration.getFolderName());
+        return folderId.get(0).toString();
+    }
+
+    private Map<TestCaseStatus, Integer> getExecutionStatuses() throws IOException
+    {
+        String json = client.executeGet(ZAPI_ENDPOINT + "util/testExecutionStatus");
+        Map<TestCaseStatus, Integer> statusIdMap = new EnumMap<>(TestCaseStatus.class);
+        zephyrExporterConfiguration.getStatuses().entrySet().forEach(s ->
+        {
+            List<Integer> statusId = JsonPathUtils.getData(json, String.format("$.[?(@.name=='%s')].id",
+                    s.getValue()));
+            notEmpty(statusId, "Status '%s' does not exist",
+                    zephyrExporterConfiguration.getFolderName());
```

Comment:
please make sure you use max line length = 120 (it's seems some lines in different code parts have unnecessary line breaks)